src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -243,7 +243,14 @@ def _find_performance_level_capability(sku, supported_service_level_objectives,
         try:
             # Find requested service objective based on capacity & family.
             # Note that for non-vcore editions, family is None.
-            return next(slo for slo in supported_service_level_objectives
+            if requesting_serverless :
+                return next(slo for slo in supported_service_level_objectives
+                        if ((slo.sku.family == sku.family) or
+                            (slo.sku.family is None and allow_reset_family)) and
+                        int(slo.sku.capacity) == int(sku.capacity) and
+                        "_S_" in slo.sku.name)
```

Comment:
Instead of copy-pasting, just change existing code to 

```
                return next(slo for slo in supported_service_level_objectives
                        if ((slo.sku.family == sku.family) or
                            (slo.sku.family is None and allow_reset_family)) and
                        int(slo.sku.capacity) == int(sku.capacity) and
                        (not requesting_serverless or "_S_" in slo.sku.name)
```

Reply:
Is "_S_" really the only way we have of distinguishing serverless? This seems fragile. Doesn't capability have compute model or min capacity specified?

Reply:
yes, S is the only differentiation